### PR TITLE
Backup wallet to file with user selection

### PR DESCRIPTION
**Pull request by yhaspel**
Save a copy of the wallet.

Screen Shot 2019-04-02 at 1 36 16 PM

### CODE REVIEW

File: app/redux/wallet/actions.js [hunk 1]
```diff
@@ -73,6 +73,19 @@ export const saveNewWallet = ({ mnemonic, salt = cryptoConsts.DEFAULT_SALT }: {
   }
 };
 
+export const saveCurrentWallet = (): Action => (dispatch: Dispatch, getState: GetState): Dispatch => {
+  const walletState = getState().wallet;
+  const { wallet, fileKey } = walletState;
+  const encryptedAccountsData = cryptoService.encryptData({ data: JSON.stringify(wallet.crypto.cipherText), key: fileKey });
```
Comment (IlyaVi) [inline app/redux/wallet/actions.js]: deep copy wallet data before you do any manipulations on it

File: app/redux/wallet/actions.js [hunk 2]
```diff
@@ -73,6 +73,19 @@ export const saveNewWallet = ({ mnemonic, salt = cryptoConsts.DEFAULT_SALT }: {
...
+  const encryptedWallet = { ...wallet, crypto: { cipher: 'AES-128-CTR', cipherText: encryptedAccountsData } };
+  try {
+    fileSystemService.saveFile({ fileContent: JSON.stringify(encryptedWallet), showDialog: true });
+    dispatch(updateWalletData({ wallet }));
```
Comment (IlyaVi) [inline app/redux/wallet/actions.js]: why calling wallet data update? No change happened during backup. Remove it

File: app/redux/wallet/actions.js [hunk 3]
```diff
@@ -73,6 +73,19 @@ export const saveNewWallet = ({ mnemonic, salt = cryptoConsts.DEFAULT_SALT }: {
   }
 };
 
+export const saveCurrentWallet = (): Action => (dispatch: Dispatch, getState: GetState): Dispatch => {
```
Comment (IlyaVi) [inline app/redux/wallet/actions.js]: rename function to describe the actual action - backupWalletToFile

File: app/screens/wallet/Backup.js [hunk 4]
```diff
@@ -192,9 +197,21 @@ class Backup extends Component<Props> {
 
   createPaperBackup = () => {};
 
-  showWalletFile = () => {};
+  showWalletFile = () => {
```
Comment (IlyaVi) [inline app/screens/wallet/Backup.js]: give showWalletFile an appropriate name

File: app/redux/wallet/actions.js (hunk 2 above)
Comment (yhaspel) [reply app/redux/wallet/actions.js]: ok

**Pull request merged**